text_generation/causal_lm/cpp/continuous_batching/python/tests/common.py

```diff
@@ -174,20 +186,62 @@ def compare_results(hf_result, ov_result, generation_config):
+
+
+def _generate_and_compare_with_reference_results(model_path: Path, prompts: List[str], reference_results: List[GenerationResult], generation_configs: List[GenerationConfig], scheduler_config: SchedulerConfig):
+    ov_results : List[GenerationResult] = run_continuous_batching(model_path, scheduler_config, prompts, generation_configs)
+
+    assert len(prompts) == len(reference_results)
+    assert len(prompts) == len(ov_results)
+
+    for prompt, ref_result, ov_result, generation_config in zip(prompts, reference_results, ov_results, generation_configs):
+        print(f"Prompt = {prompt}\nref result = {ref_result}\nOV result = {ov_result}")
+        compare_results(ref_result, ov_result, generation_config)
+
+
+def generate_and_compare_with_reference_text(model_path: Path, prompts: List[str], reference_texts_per_prompt: List[List[str]], generation_configs: List[GenerationConfig], scheduler_config: SchedulerConfig):
+    ov_results : List[GenerationResult] = run_continuous_batching(model_path, scheduler_config, prompts, generation_configs)
+
+    assert len(prompts) == len(reference_texts_per_prompt)
+    assert len(prompts) == len(ov_results)
+
+    for prompt, ref_texts_for_this_prompt, ov_result, generation_config in zip(prompts, reference_texts_per_prompt, ov_results, generation_configs):
+        print(f"Prompt = {prompt}\nref text = {ref_texts_for_this_prompt}\nOV result = {ov_result}")
+
+        assert len(ref_texts_for_this_prompt) == len(ov_result.m_generation_ids)
+        for ref_text, ov_text in zip(ref_texts_for_this_prompt, ov_result.m_generation_ids):
+            assert ref_text == ov_text
 
 def run_test_pipeline(tmp_path: str, model_id: str, scheduler_params: dict = None):
     prompts, generation_configs = get_test_dataset()
     scheduler_config = get_scheduler_config(scheduler_params)
 
-    (hf_results, model_path) = run_hugging_face(model_id=model_id, prompts=prompts,
-                                                generation_configs=generation_configs, tmp_path=tmp_path,
-                                                use_optimum=True)
-    ov_results: List[GenerationResult] = run_continuous_batching(model_path, scheduler_config, prompts,
-                                                                 generation_configs)
+    _generate_and_compare_with_hf(model_id, prompts, generation_configs, scheduler_config, tmp_path)
 
-    assert len(prompts) == len(hf_results)
-    assert len(prompts) == len(ov_results)
 
-    for prompt, hf_result, ov_result, generation_config in zip(prompts, hf_results, ov_results, generation_configs):
-        print(f"Prompt = {prompt}\nHF result = {hf_result}\nOV result = {ov_result}")
-        compare_results(hf_result, ov_result, generation_config)
+DEFAULT_SCHEDULER_CONFIG = get_scheduler_config({"num_kv_blocks": 300, "block_size": 16, "dynamic_split_fuse": True, "max_num_batched_tokens": 256, "max_num_seqs": 256})
```

Comment:
```suggestion
DEFAULT_SCHEDULER_CONFIG = get_scheduler_config({"num_kv_blocks": 300, "dynamic_split_fuse": True, "max_num_batched_tokens": 256, "max_num_seqs": 256})
```
In latest OpenVINO `master`, CPU plugin supports only block size 32

Reply:
Done